packages/wdio-crossbrowsertesting-service/src/service.ts

```diff
@@ -0,0 +1,218 @@
+import got from 'got'
+import logger from '@wdio/logger'
+
+const log = logger('@wdio/crossbrowsertesting-service')
+const jobDataProperties = ['name', 'tags', 'public', 'build', 'extra']
+
+export default class CrossBrowserTestingService implements WebdriverIO.ServiceInstance {
+    private _browser!: WebdriverIO.BrowserObject | WebdriverIO.MultiRemoteBrowserObject
+    private _testCnt = 0;
+    private _failures = 0;
+    private _capabilities!: WebDriver.DesiredCapabilities;
+    private _config?: WebdriverIO.Options;
+    private _isServiceEnabled?: boolean;
+    private _suiteTitle?: string;
+    private _cbtUsername?: string;
+    private _cbtAuthkey?: string;
+
+    /**
+     * gather information about runner
+     */
+    beforeSession (config: WebdriverIO.Options, capabilities: WebDriver.DesiredCapabilities) {
```

Comment:
See if you could do all these assignments in the constructor (it should have access to config, capabilities and options). If that works we can make all these properties required, e.g.

```patch
-    private _cbtUsername?: string;
-    private _cbtAuthkey?: string;
+    private _cbtUsername: string;
+    private _cbtAuthkey: string;
```

Reply:
So shall we remove the `beforeSession` block altogether?

Reply:
yes